hangman_edited_1

```diff
@@ -0,0 +1,141 @@
+# Name:
+# Section: 
+# 6.189 Project 1: Hangman template
+# hangman_template.py
+
+# Import statements: DO NOT delete these! DO NOT write code above this!
+from random import randrange
+from string import *
+from hangman_lib import print_hangman_image
+import os
+
+# -----------------------------------
+# Helper code
+# (you don't need to understand this helper code)
+# Import hangman words
+
+WORDLIST_FILENAME = "words.txt"
+
+def load_words():
+    """
+    Returns a list of valid words. Words are strings of lowercase letters.
+    
+    Depending on the size of the word list, this function may
+    take a while to finish.
+    """
+    print ("Loading word list from file...")
+    # inFile: file
+    inFile = open(WORDLIST_FILENAME, 'r')
+    # line: string
+    line = inFile.readline()
+    # wordlist: list of strings
+    wordlist = line.split()
+    print ("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+# actually load the dictionary of words and point to it with 
+# the words_dict variable so that it can be accessed from anywhere
+# in the program
+words_dict = load_words()
+
+
+# Run get_word() within your program to generate a random secret word
+# by using a line like this within your program:
+# secret_word = get_word()
+
+def get_word():
+    """
+    Returns a random word from the word list
+    """
+    word=words_dict[randrange(0,len(words_dict))]
+    return word
+
+# end of helper code
+# -----------------------------------
+# From part 3b:
+def word_guessed():
+    '''
+    Returns True if the player has successfully guessed the word,
+    and False otherwise.
+    '''
+    global secret_word
+    global letters_guessed
+
+    ####### YOUR CODE HERE ######
+    for letter in secret_word:
+        if letter not in letters_guessed:
+            return False
+    return True
+
+def print_guessed():
+    '''
+    Prints out the characters you have guessed in the secret word so far
+    '''
+    global secret_word
+    global letters_guessed
+    
+    ####### YOUR CODE HERE ######
+    print_letters = ""
+    for letter in secret_word:
+        if letter in letters_guessed:
+            print_letters += letter
+        else:
+            print_letters += "-"
+    return print_letters
+
+def is_input_letter_valid(input_letter):
+    if input_letter not in letters_guessed:
+        if input_letter.isalpha() and len(input_letter) == 1:
+            return True
+        else:
+            print("Please enter only one alphabetical letter.")
+            return False
+    else:
```

Comment:
Inverting this if-else statements would make it easier to read (1 less nested if level).
e.g.:
```
This method `if` `else` could be made more readable by rearranging them.
For example:
```
if input_letter in letters_guessed:
  print("Guessed already.Choose another.")
  return False

if input_letter.isalpha() and len(input_letter) == 1:
  [...]
else:
  [...]
```